threedi_schema/migrations/versions/0300_upgrade_db_settings.py

```diff
@@ -0,0 +1,296 @@
+    # no checks for existence are done, this will fail if any table or column doesn't exist
+    dst_cols = ', '.join(dst for _, dst in columns)
+    src_cols = ', '.join(src for src, _ in columns)
+    op.execute(sa.text(f'INSERT INTO {dst_table} ({dst_cols}) SELECT {src_cols} FROM {src_table}'))
+    remove_columns_from_table(src_table, [src for src, _ in columns])
+
+
+def move_values_to_table(src_table: str, src_col: str, dst_table: str, dst_col: str):
+    op.execute(f'UPDATE {dst_table} SET {dst_col} = (SELECT {src_col} FROM {src_table} LIMIT 1)')
+    remove_columns_from_table(src_table, [src_col])
+
+
+def remove_columns_from_table(table_name: str, columns: List[str]):
+    # no checks for existence are done, this will fail if any table or column doesn't exist
+    with op.batch_alter_table(table_name) as batch_op:
+        for column in columns:
+            batch_op.drop_column(column)
+
+
+def set_use_from_settings_id():
+    for settings_col, settings_id, settings_table in GLOBAL_SETTINGS_ID_TO_BOOL:
+        if settings_col != '':
+            # set boolean 'use_*' in model_settings if a relationship exists
+            op.execute(f"UPDATE model_settings SET {settings_col} = TRUE WHERE {settings_id} IS NOT NULL;")
+        # remove all settings rows, exact for the one matching
+        # delete_all_but_matching_id(settings_table, settings_id)
+        op.execute(
+            f"DELETE FROM {settings_table} WHERE NOT EXISTS "
+            f"(SELECT 1 FROM model_settings WHERE model_settings.{settings_id} IS NOT NULL);")
+        # command above doesn't catch id = None, so an extra command is needed for those cases
+        op.execute(
+            f"DELETE FROM {settings_table} WHERE id != "
+            f"(SELECT {settings_id} FROM model_settings);")
+
+    # set use_groundwater_flow
+    sql = """
+        UPDATE model_settings
+        SET use_groundwater_flow = CASE
+            WHEN (SELECT groundwater_hydraulic_conductivity FROM groundwater LIMIT 1) IS NOT NULL OR (SELECT groundwater_hydraulic_conductivity_file FROM groundwater LIMIT 1) IS NOT NULL THEN 1
+            ELSE 0
+        END;
+        """
+    op.execute(sql)
+
+
+def drop_columns(table, columns):
+    with op.batch_alter_table(table) as batch_op:
+        for column in columns:
+            batch_op.drop_column(column)
+
+
+def set_use_from_exists(table, settings_col):
+    op.execute(f"UPDATE model_settings SET {settings_col} = TRUE WHERE EXISTS (SELECT 1 FROM {table});")
+
+
+def delete_all_but_matching_id(table, settings_id):
+    op.execute(f"DELETE FROM {table} WHERE id NOT IN (SELECT {settings_id} FROM model_settings);")
+
+
+def delete_all_but_first_row(table):
+    op.execute(f"DELETE FROM {table} WHERE id NOT IN ("
+               f"SELECT id FROM "
+               f"(SELECT id FROM {table} ORDER BY id LIMIT 1) AS subquery);")
+
+
+def correct_dem_paths():
+    # remove path - this will not work with nested paths!
```

Comment:
Nested paths definitely occur.

I recommend doing this one in python, using `pathlib.Path(...).name` to match https://github.com/nens/threedi-modelchecker/blob/7664e375944077095af51ce9b7479081ab034662/threedi_modelchecker/checks/raster.py#L70

Reply:
See my comment in the modelchecker PR; modelchecker does check for file presence using the path! The upgrade there has to match the migration here.

Reply:
Changed setting dem path in af492fe08d1a8b29501d8566311a8766ed189f52

Reply:
Fixed handling of paths in modelchecker: 4acf4dd4228e83eeb6d2afb59a3a4089f13339b3.